proxy/main.go

```diff
@@ -21,31 +26,45 @@ func main() {
 		os.Exit(1)
 	}
 
-	defer listener.Close()
-
-	for {
-		// Listen for an incoming connection.
-		conn, err := listener.Accept()
-		if err != nil {
-			fmt.Println("Error accepting: ", err.Error())
-			os.Exit(1)
-		}
-		// Handle connections in a new goroutine.
-		go handleRequest(conn)
+	server := http.Server{
+		Handler: http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			handleHTTP(w, r)
+		}),
 	}
+
+	server.Serve(listener)
 }
 
-// Handles incoming requests.
-func handleRequest(conn net.Conn) {
-	// Make a buffer to hold incoming data.
-	buf := make([]byte, 1024)
-	// Read the incoming connection into the buffer.
-	_, err := conn.Read(buf)
+func handleHTTP(w http.ResponseWriter, req *http.Request) {
+	requestDump, err := httputil.DumpRequest(req, true)
+	if err != nil {
+		fmt.Println(err)
+		return
+	} else {
+		fmt.Println(string(requestDump))
+	}
+
+	u, err := url.Parse("http://localhost:9994/")
```

Comment:
We need to replace req.URL here, even when we address the request to the http://localhost:9994 destination in the client.  I'm not entirely sure why.